openupgradelib/openupgrade_merge_records.py

```diff
@@ -431,6 +431,111 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
         )
 
 
+def _adjust_merged_values_sql(env, model_name, record_ids, target_record_id,
+                              exclude_columns, model_table, field_spec):
+    """This method deals with the values on the records to be merged +
+    the target record, performing operations that makes sense on the meaning
+    of the model.
+
+    :param field_spec: Dictionary with field names as keys and forced operation
+      to perform as values. If a field is not present here, default operation
+      will be performed.
+
+      Possible operations by field types same as _adjust_merged_values_orm.
+    """
+    if not column_exists(env.cr, model_table, 'id'):
+        return
+    env.cr.execute(
+        """
+        SELECT isc.column_name, isc.data_type, imf.ttype
+        FROM information_schema.columns isc
+        JOIN ir_model_fields imf ON (
+            imf.name = isc.column_name AND imf.model = %s)
+        WHERE table_name = %s
+        """, (model_table, model_name))
+    vals = {}
+    new_vals = {}
+    for column, column_type, field_type in env.cr.fetchall():
+        if (model_table, column) in exclude_columns:
+            continue
+        op = field_spec.get(column, False)
+        env.cr.execute(
+            sql.SQL(
+                """SELECT {column}
```

Comment:
You are doing one SELECT per column. This is not performance-wise. Read all the columns in one shot.